form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/FormFlowResource.kt

```diff
@@ -40,41 +43,35 @@ class FormFlowResource(
 
         val stepInstance = instance.getCurrentStep()
 
-        return ResponseEntity.ok(
-            GetFormFlowStateResult(
-                instance.id.id,
-                FormFlowStepResult(
-                    stepInstance.id.id,
-                    stepInstance.definition.type.name,
-                    stepInstance.definition.type.properties
-                    // TODO Include the prefilled form when this functionality is available
-                )
-            )
-        )
+        return ResponseEntity.ok(GetFormFlowStateResult(instance.id.id, getStepResult(stepInstance)))
     }
 
-    @PostMapping("/{formFlowId}/step/{stepInstanceId}")
+    @PostMapping("/{formFlowId}/step/{stepInstanceId}", produces = [MediaType.APPLICATION_JSON_VALUE])
+    @Transactional
     fun completeStep(
         @PathVariable(name = "formFlowId") formFlowId: String,
         @PathVariable(name = "stepInstanceId") stepInstanceId: String,
         @RequestBody submissionData: JsonNode?
-    ): ResponseEntity<FormFlow> {
-        return ResponseEntity.ok(getStepDto())
+    ): ResponseEntity<CompleteStepResult> {
+        val instance = formFlowService.getByInstanceIdIfExists(
+            FormFlowInstanceId.existingId(UUID.fromString(formFlowId))
+        )!!
+
+        val stepInstance = instance.complete(
+            FormFlowStepInstanceId.existingId(UUID.fromString(stepInstanceId)),
+            JSONObject((submissionData ?: JsonNodeFactory.instance.objectNode()).toString())
+        )!!
+
+        return ResponseEntity.ok(CompleteStepResult(instance.id.id, getStepResult(stepInstance)))
     }
 
-    private fun getStepDto(): FormFlow {
-        return FormFlow(
-            UUID.randomUUID(),
-            FormFlowStep(
-                UUID.randomUUID(),
-                "form",
-                FormTypeProperties(
-                    formDefinitionService
-                        .getFormDefinitionByName("user-task-lening-aanvragen")
-                        .get()
-                        .formDefinition
-                )
-            )
+    private fun getStepResult(stepInstance: FormFlowStepInstance): FormFlowStepResult {
+        return FormFlowStepResult(
+            stepInstance.id.id,
+            stepInstance.definition.type.name,
+            jacksonObjectMapper().createObjectNode()
+                .set("definition", jacksonObjectMapper().createObjectNode().put("firstName", "John"))
```

Comment:
I assume this is temporary code?

Reply:
yes, definitely.
Its basically to keep the old mock-endpoint alive for a bit. Since the mock-endpoint returned the formIo of 'user-task-lening-aanvragen'.

Reply:
Not entirely sure how this relates to the demo endpoint. Can you elaborate?

Reply:
The mock-endpoints were  introduced so that the FE can already start developing. So its different from the demo-endpoints